### PR TITLE
Car Insurance Prediction

### PR DESCRIPTION
**Pull request by soma2000-lang**
## Related Issue 

- Info about the related issue 

- [x] LGM-SOC'21 Participant 
- [x] Contributor

Closes: #463

### Descrition

PROJECT TITLE : Car insurance prediction

GOAL : To predict the insurance  in cars
DATASET : 
WHAT I HAD DONE : In this project first I performed a exploratory data analysis on the  dataset which includes of data cleaning , data manipulation, data preprocessing , data visualization and after that I did the model building using  machine learning classification and regression algorithms and then predicted the accuracy of  the model . In the model prediction part I used different machine learning algorithms  . In the EDA part I have included different plots for the different visualizations of our dataset . During the model prediction I got different accuracies from different models , I got the highest accuracy of 99 % using the Random Forest Regressor which is quite well for the given Brazil Fires dataset . Some plots which I used for visualizing the dataset are Histogram., Boxplot, Heatmap etc.

LIBRARIES:

PANDAS

NUMPY

MATPLOTLIB

TENSORFLOW

SKLEARN

CONCLUSION :

So we get a good accuracy and training score of about 99% using Random Forest Regressor

## Type of change

What sort of change have you made:

- [x] My code follows the guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly wherever it was hard to understand.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added things that prove my fix is effective or that my feature works.

### CODE REVIEW

**Comment by soma2000-lang** [comment]
image

@prathimacode-hub  I have changed rom model to Model but dont know Why its not reflecting here

**Comment by prathimacode-hub** [comment]
It's because it wasn't cloned properly in your local drive. Why did you create new PR? @soma2000-lang

**Comment by soma2000-lang** [comment]
> It's because it wasn't cloned properly in your local drive. Why did you create new PR? @soma2000-lang

@prathimacode-hub mam I dont know why did this happened ,I forked it and then cloned it only.Mam since I was unable to reopen the closed pr
that why I just created a new branch and gave another pr

**Comment by prathimacode-hub** [comment]
Ok, First thing to be done is you have to change the template in PR. You had added feature request isue template instead of PR template. Kindly change that and add the details of your project and your issue number accordingly. @soma2000-lang

**Review by prathimacode-hub — CHANGES_REQUESTED** [review]
Changes to be done:
The project title should be as "Car Insurance Prediction"
The link of the dataset you had added is not an opening link. I guess this is the dataset you had used. (https://www.kaggle.com/infernape/fast-furious-and-insured). Kindly change it to this link. Not of the link you added up in your local drive.
The model folder should be titled as "Model"
In 'README.md' elaborate the goal and also in accuracies, list out all the models you had used along with their accuracies appropriately.

Kindly do these changes needed.

And as you said, you changed it in your drive but isn't showing up the changes here in main repo. The reason is proper worklow isn't followed or not cloned properly into your local drive. If you want I can share you with the easy process to make this work. But for that, you need to remove the forked repository from your profile and refork it again. Before that, make sure you have all your files updated in your local drive of your system. 

@soma2000-lang

**Comment by prathimacode-hub** [comment]
Don't create PR again. We can make it work with this PR itself. @soma2000-lang

**Comment by soma2000-lang** [comment]
> Don't create PR again. We can make it work with this PR itself. And one more thing, you should neevr close issues or PR from your end. @soma2000-lang

I did not do it intentionlly,somehow it got

**Comment by soma2000-lang** [comment]
> In 'README.md' elaborate the goal and also in accuracies, list out all the models you had used along with their accuracies appropriately.
> 

> 
> And as you said, you changed it in your drive but isn't showing up the changes here in main repo. The reason is proper worklow isn't followed or not cloned properly into your local drive. If you want I can share you with the easy process to make this work. But for that, you need to remove the forked repository from your profile and refork it again. Before that, make sure you have all your files updated in your local drive of your system.
> 
> @soma2000-lang
Mam actually I did only one model hemce only accuracy of only one model which is 99% .

Yes I have my file in my local system and I am deleting the dorked repo and reforking it again

**Comment by prathimacode-hub** [comment]
> > Don't create PR again. We can make it work with this PR itself. And one more thing, you should neevr close issues or PR from your end. @soma2000-lang
> 
> I did not do it intentionlly,somehow it got.Sorry for that

Sorry isn't needed, I am just letting you know so that it doesn't happen in future. I have no issue with it but not all would take this positive

**Comment by prathimacode-hub** [comment]
If you had done one model, then add that up, but it's better if you could atleast use min of 3 models and compare the efficiency. That way model would be more optimized. @soma2000-lang

**Comment by soma2000-lang** [comment]
@I have deleted the fork and reforked it again

**Comment by prathimacode-hub** [comment]
> @prathimacode-hub I have deleted the fork and reforked it again

Ok, you need help in making it proper? For that first, do all the changes asked in files in local drive and keep it ready. @soma2000-lang

**Comment by soma2000-lang** [comment]
> > @prathimacode-hub I have deleted the fork and reforked it again
> 
> Ok, you need help in making it proper? For that first, do all the changes asked in files in local drive and keep it ready. @soma2000-lang

Yes Mam After that what is to be done

**Comment by soma2000-lang** [comment]
@prathimacode-hub  I am making an Images folder also which has all the visualization images what is the naming format of each png image?? CAn i name it as image1 ,image2 like that??

**Comment by soma2000-lang** [comment]
> @prathimacode-hub I am making an Images folder also which has all the visualization images what is the naming format of each png image?? CAn i name it as image1 ,image2 like that??
@prathimacode-hub  please reply

**Comment by Manasi2001** [comment]
> @prathimacode-hub I am making an Images folder also which has all the visualization images what is the naming format of each png image?? CAn i name it as image1 ,image2 like that??

Yes @soma2000-lang, you can name the images like that.

Do you need help with anything else?

**Comment by soma2000-lang** [comment]
> > @prathimacode-hub I have deleted the fork and reforked it again
> 
> Ok, you need help in making it proper? For that first, do all the changes asked in files in local drive and keep it ready. @soma2000-lang

@prathimacode-hub   I have finally have my files ready with all my changes made now Should Ihave to clone the repo again,I have deleted the fork and reforked it as u told me to do

**Comment by prathimacode-hub** [comment]
> > > @prathimacode-hub I have deleted the fork and reforked it again
> > 
> > 
> > Ok, you need help in making it proper? For that first, do all the changes asked in files in local drive and keep it ready. @soma2000-lang
> 
> @prathimacode-hub I have finally have my files ready with all my changes made now Should Ihave to clone the repo again,I have deleted the fork and reforked it as u told me to do

Great, instead of cloning we can directly create PR through the forked respository itself. Text me when you are here. I shall guide you on it. @soma2000-lang

**Comment by soma2000-lang** [comment]
> > > > @prathimacode-hub I have deleted the fork and reforked it again
> > > 
> > > 
> > > Ok, you need help in making it proper? For that first, do all the changes asked in files in local drive and keep it ready. @soma2000-lang
> > 
> > 
> > @prathimacode-hub I have finally have my files ready with all my changes made now Should Ihave to clone the repo again,I have deleted the fork and reforked it as u told me to do
> 
> Great, instead of cloning we can directly create PR through the forked respository itself. Text me when you are here. I shall guide you on it. @soma2000-lang

Yes please say @prathimacode-hub .I have texted u on discord mam

**Comment by prathimacode-hub** [comment]
I shall provide a discussion link with all the steps to follow for the changes you need in your project by today night. @soma2000-lang

**Comment by prathimacode-hub** [comment]
Here is the discussion link I had created for you to refer and automate your process with the changes. Look at it : https://github.com/prathimacode-hub/ML-ProjectKart/discussions/508  

@soma2000-lang
